realm/realm-library/src/main/cpp/io_realm_ObjectStoreUserStore.cpp

```diff
@@ -0,0 +1,150 @@
+
+static const std::shared_ptr<SyncUser>& currentUserOrThrow(JNIEnv *env) //throws
+{
+    std::vector<std::shared_ptr<SyncUser>> all_users = SyncManager::shared().all_users();
+    if (all_users.size() > 1) {
+        throw std::runtime_error(ERR_MULTIPLE_LOGGED_IN_USERS);
+    } else if (all_users.size() < 1) {
+        throw std::runtime_error(ERR_NO_LOGGED_IN_USER);
+    } else {
+        return all_users.front();
+    }
+}
+
+/*
+ * Class:     io_realm_ObjectStoreUserStore
+ * Method:    getCurrentUser
+ * Signature: ()Ljava/lang/String;
+ */
+JNIEXPORT jstring JNICALL Java_io_realm_ObjectStoreUserStore_getCurrentUser (JNIEnv *env, jclass)
+{
+    TR_ENTER()
+    try {
+        const std::shared_ptr<SyncUser> &user = currentUserOrThrow(env);
+        if (user->state() == SyncUser::State::Active) {
+            return to_jstring(env, user->refresh_token().data());
+        } else {
+            return NULL;
+        }
+    } CATCH_STD()
+}
+
+/*
+ * Class:     io_realm_ObjectStoreSyncManager
+ * Method:    updateOrCreateUser
+ * Signature: (Ljava/lang/String;Ljava/lang/String;)V
+ */
+JNIEXPORT void JNICALL Java_io_realm_ObjectStoreUserStore_updateOrCreateUser (JNIEnv *env, jclass, jstring identity, jstring jsonToken, jstring url)
+{
+    TR_ENTER()
+    try {
+        JStringAccessor user_identity(env, identity); // throws
+        JStringAccessor user_json_token(env, jsonToken); // throws
+        JStringAccessor auth_url(env, url); // throws
+
+        SyncManager::shared().get_user(user_identity, user_json_token, util::Optional<std::string>(auth_url));
+    } CATCH_STD()
+}
+
+/*
+ * Class:     io_realm_ObjectStoreSyncManager
+ * Method:    logoutCurrentUser
+ * Signature: ()V
+ */
+JNIEXPORT void JNICALL Java_io_realm_ObjectStoreUserStore_logoutCurrentUser (JNIEnv *env, jclass)
+{
+    TR_ENTER()
+    try {
+        const std::shared_ptr<SyncUser>& user = currentUserOrThrow(env);
+        user->log_out();
+    } CATCH_STD()
+}
+
+
+/*
+ * Class:     io_realm_ObjectStoreSyncManager
+ * Method:    configureMetaDataSystem
+ * Signature: (Ljava/lang/String;[B)V
+ */
+JNIEXPORT void JNICALL Java_io_realm_ObjectStoreUserStore_configureMetaDataSystem (JNIEnv *env, jclass, jstring baseFile)
+{
+    TR_ENTER()
+    try {
+        JStringAccessor base_file_path(env, baseFile); // throws
+        SyncManager::shared().configure_file_system(base_file_path, SyncManager::MetadataMode::NoEncryption);
+    } CATCH_STD()
+}
+
+/*
+ * Class:     io_realm_ObjectStoreSyncManager
+ * Method:    getAllUsers
+ * Signature: ()[Ljava/lang/String;
+ */
+JNIEXPORT jobjectArray JNICALL Java_io_realm_ObjectStoreUserStore_getAllUsers (JNIEnv *env, jclass)
+{
+    TR_ENTER()
+    std::vector<std::shared_ptr<SyncUser>> all_users = SyncManager::shared().all_users();
+    if (!all_users.empty()) {
+        jsize array_length = 0;
+        std::vector<std::shared_ptr<SyncUser>>::iterator valid_users = std::find_if(all_users.begin(),all_users.end(),
```

Comment:
this function logic doesn't seem to be correct. `find_if` will stop when it finds the first element matches the condition, so the `array_length` will be always 1(or 0) here. And the following `for_each` loop will iterate from the `valid_users` until the end of original vector, means you will get a `array_length == -1` in the loop. 

I think you can do something like:
```cpp
array_length = std::count_if(all_users.begin(),all_users.end(), [&] {...};
for (auto user : all_users) {
    if (user->state() == SyncUser::State::Active) {
         // Set the java array.
    }
}
```